x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/RelyingPartyConfiguration.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.security.authc.oidc;
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.Strings;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Class that contains all the OpenID Connect Relying Party configuration
+ */
+public class RelyingPartyConfiguration {
+    private final String clientId;
+    private final String redirectUri;
+    private final String responseType;
+    private final List<String> requestedScopes;
+
+    public RelyingPartyConfiguration(String clientId, String redirectUri, String responseType, @Nullable List<String> requestedScopes) {
+        this.clientId = Objects.requireNonNull(clientId, "RP Client ID must be provided");
+        this.redirectUri = Objects.requireNonNull(redirectUri, "RP Redirect URI must be provided");
+        if (Strings.hasText(responseType) == false) {
+            throw new IllegalArgumentException("Response type must be provided");
```

Comment:
I think these error messages need to be clear about the `Setting` that was incorrect.
Ideally we'd do validation in the `Setting` itself, but if not, then I think the validation ought to be done in the context of the Realm so that it can reference the setting key.
Otherwise we'll get lots of questions from users about "I got an error saying `Response type must be provided` but I don't know where to configure it."

Reply:
Makes sense Tim, I'll address that. I'll do it in the Realm (this is what we do in other realms too I think) but I can move this to the Setting with a validator if need be !